internal/environments/environments.go

```diff
@@ -0,0 +1,110 @@
+package environments
+
+import (
+	"github.com/trento-project/trento/internal/consul"
+	"github.com/trento-project/trento/internal/hosts"
+)
+
+type Environment struct {
+	Name       string                `mapstructure:"name,omitempty"`
+	Landscapes map[string]*Landscape `mapstructure:"landscapes,omitempty"`
+}
+
+type Landscape struct {
+	Name       string                `mapstructure:"name,omitempty"`
+	SAPSystems map[string]*SAPSystem `mapstructure:"sapsystems,omitempty"`
+}
+
+type SAPSystem struct {
+	Name  string         `mapstructure:"name,omitempty"`
+	Type  string         `mapstructure:"type,omitempty"`
+	Hosts hosts.HostList `mapstructure:"hosts,omitempty"`
+}
+
+func NewEnvironment(env, land, sys string) Environment {
+	newSys := &SAPSystem{Name: sys}
+	systemsMap := make(map[string]*SAPSystem)
+	systemsMap[sys] = newSys
+
+	newLand := &Landscape{Name: land, SAPSystems: systemsMap}
+	landsMap := make(map[string]*Landscape)
+	landsMap[land] = newLand
+
+	newEnv := Environment{Name: env, Landscapes: landsMap}
+	return newEnv
+}
+
+func ungrouped(name string) bool {
+	return name == consul.KvUngrouped
+}
+
+func (e *Environment) Ungrouped() bool {
+	return ungrouped(e.Name)
+}
+
+func (l *Landscape) Ungrouped() bool {
+	return ungrouped(l.Name)
+}
+
+func (s *SAPSystem) Ungrouped() bool {
+	return ungrouped(s.Name)
+}
```

Comment:
I'm not quite clear why do we need this `ungrouped` fake group. Can we do without it somehow?
Maybe we could just assign every node to a `default` environment and landscape, and treat it like any other actual value. I think there is no need to have a special one with conditional around it: we just need to assign one by default.
This would also have the advantage that in the absence of user input, we can always show something in the environments and landscape lists anyway.

Reply:
This is a business decision I guess. I'm fine with any of the options. @lee-martin What do you think?
As a context, now new discovered `SAP systems` are added to a environmend and landscape named `ungrouped`, which means that they don't belong to any particular env/land. I use this to have a different treatment in the frontend. 
If you prefer a default Env/Land it is fine

Reply:
@lee-martin are you okay with displaying Systems under a `default` Environment and Landscape, in the absence of user input?

Reply:
@arbulu89  & @stefanotorresi  oh wow, this is quite difficult, I can also see both points of view here as well. I would go with @stefanotorresi and the idea to put hosts into some default environment and landscape here, since I agree that the empty pages would indeed be confusing.  What would be the name of the default environment and landscapebe? just call it "default" ?
In any event, we need to see and use it, then we can iterate and improve.

Reply:
Yes, my idea is to just call it `default`, and then in the future we give users the ability to rename it and/or create new ones and reassign systems to them.

Reply:
Ok, let's do it out of this PR, as it would require changes in many files (specially frontend templates) and It would grow up much more the PR

Reply:
All right! I will add a follow-up issue for this.